Library/Homebrew/cmd/shellenv.sh

```diff
@@ -6,16 +6,16 @@
 #:  Consider adding evaluation of this command's output to your dotfiles (e.g. `~/.profile`, `~/.bash_profile`, or `~/.zprofile`) with: `eval $(brew shellenv)`
 
 homebrew-shellenv() {
-  case "$SHELL" in
-    */fish|fish)
+  case "$(/bin/ps -p $PPID -o comm=)" in
```

Comment:
```suggestion
  case "$(/bin/ps -hp $$ -o comm=)" in
```

Reply:
That would always return `bash` (the `brew` process itself).

Reply:
It passes for me:
```
$ zsh
mbelkin@Gaia ~ % /bin/ps -p $$ -o comm=
zsh
mbelkin@Gaia ~ % 
```

Reply:
but neither your solution nor mine work in fish

Reply:
oh, got it! you're right I was trying to call it using different shells instead of calling it using bash. my bad!